[doc] Add `-pthread` to the multithreaded tutorial test (#1870)
eth-cscs/reframe: docs/tutorial_basics.rst

==========================================
 Tutorial 1: Getting Started with ReFrame
==========================================

.. versionadded:: 3.1

This tutorial will give you a first overview of ReFrame and will acquaint you with its basic concepts.
We will start with a simple "Hello, World!" test running with the default configuration and we will expand the example along the way.
We will also explore performance tests and port our tests to an HPC cluster.

> The examples of this tutorial can be found under :obj:`tutorials/basics/`.

Reviewer: I think that you have broken something here in the formating.

   :language: cpp
   :lines: 6-

> This program takes as argument the number of threads it will create and it uses ``std::thread``, which is C++11 addition, meaning that we will need to pass ``-std=c++11``, as well as ``-pthread`` to our compilers.

Reviewer: ```suggestion
This program takes as argument the number of threads it will create and it uses ``std::thread``, which is C++11 addition, meaning that we will need to pass ``-std=c++11`` and ``-pthread`` to our compilers.
```

   :lines: 6-
   :emphasize-lines: 11-13

In order to compile applications using ``std::thread`` with GCC and Clang, the ``-pthread`` option has to be passed to the compiler.
Since the above option might not be valid for other compilers, we use pipeline hooks to differentiate based on the programming environment as follows:

.. code-block:: python

   @rfm.run_before('compile')

>    def setpthreadsflag(self):

Reviewer: ```suggestion
   def set_threading_flags(self):
```

       environ = self.current_environ.name
       if environ in {'clang', 'gnu'}:
           self.build_system.cxxflags += ['-pthread']

.. note::

>    The pipeline hooks are covered in more detail in a following tutorial section.

Reviewer: ```suggestion
   The pipeline hooks, as well as the regression test pipeline itself, are covered in more detail later on in the tutorial.
```

>    For a detailed coverage of the regression test pipeline, please refer to :doc:`pipeline`.

Reviewer: I wouldn't put this sentence here. The pipeline is covered in the discussion about the hooks.

ReFrame delegates the compilation of a test to a *build system*, which is an abstraction of the steps needed to compile the test.
Build systems take also care of interactions with the programming environment if necessary.
Compilation flags are a property of the build system.
If not explicitly specified, ReFrame will try to pick the correct build system (e.g., CMake, Autotools etc.) by inspecting the test resources, but in cases as the one presented here where we need to set the compilation flags, we need to specify a build system explicitly.

> In this example, we instruct ReFrame to compile a single source file using the ``-std=c++11 -pthread -Wall`` (for GCC and Clang).

Reviewer: The original text is fine here, I think.
